public/pages/Destinations/containers/CreateDestination/ManageEmailGroups/ManageEmailGroups.js

```diff
@@ -0,0 +1,261 @@
+      emails: updatedEmailGroup.emails.map(email => ({ email: email.label })),
+    };
+    try {
+      await httpClient.put(
+        `../api/alerting/email_groups/${id}?ifSeqNo=${ifSeqNo}&ifPrimaryTerm=${ifPrimaryTerm}`,
+        body
+      );
+    } catch (err) {
+      console.error('Unable to update email group', err);
+    }
+  };
+
+  deleteEmailGroup = async emailGroup => {
+    const { httpClient } = this.props;
+    const { id } = emailGroup;
+    try {
+      await httpClient.delete(`../api/alerting/email_groups/${id}`);
+    } catch (err) {
+      console.err('Unable to delete email group', err);
+    }
+  };
+
+  // TODO: Cleanup this function (currently making sequential API calls since each one has 'awaits' on it)
+  processEmailGroups = async values => {
+    const { emailGroupsToDelete } = this.state;
+    const { emailGroups } = values;
+
+    // Create or update email groups
+    for (const emailGroup of emailGroups) {
+      if (emailGroup.state === STATE.CREATED) {
+        await this.createEmailGroup(emailGroup);
+      } else if (emailGroup.state === STATE.UPDATED) {
+        await this.updateEmailGroup(emailGroup);
+      }
+    }
+
+    // Delete any removed email groups
+    for (const emailGroup of emailGroupsToDelete) {
+      await this.deleteEmailGroup(emailGroup);
+    }
+  };
+
+  renderEmailGroups = ({ values, arrayHelpers }) => {
+    const hasEmailGroups = !_.isEmpty(values.emailGroups);
+    // TODO: Change this to getEmailOptions from stored emailGroups state
+    const emailOptions = getEmailOptions(values.emailGroups);
+    return hasEmailGroups ? (
+      <div>
+        {values.emailGroups.map((emailGroup, index) => (
+          <div key={index}>
+            <EmailGroup
+              emailGroup={emailGroup}
+              emailOptions={emailOptions}
+              arrayHelpers={arrayHelpers}
+              context={createEmailGroupContext(values.emailGroups)}
+              index={index}
+              onDelete={() => {
+                if (emailGroup.id) {
+                  this.setState(prevState => ({
+                    emailGroupsToDelete: [...prevState.emailGroupsToDelete, emailGroup],
+                  }));
+                }
+                arrayHelpers.remove(index);
+              }}
+            />
+            <EuiSpacer className="accordion-separator" />
+          </div>
+        ))}
+        <div style={{ justifyContent: 'left' }}>
+          <AddEmailGroupButton arrayHelpers={arrayHelpers} />
+        </div>
+      </div>
+    ) : (
+      <EmailGroupEmptyPrompt arrayHelpers={arrayHelpers} />
+    );
+  };
+
+  render() {
+    const { isVisible, onClickCancel, onClickSave } = this.props;
+    const { initialValues, loadingEmailGroups, emailGroupsToDelete } = this.state;
+    return isVisible ? (
+      <Formik
+        initialValues={initialValues}
+        onSubmit={(values, formikBag) => {
+          this.processEmailGroups(values).then(() => onClickSave());
```

Comment:
Seems like even if process email groups fail it still closes everything?

Reply:
Since multiple email groups are being processed, we try to process them all and still close even if a couple fail because the action that is taken is based on them being tagged `CREATED/UDPATED/DELETED` so if a created sender succeeds and another fails and the form is left open, it will still be marked as `CREATED` and will attempt to call a `POST` API on an existing sender when it's submitted again which is incorrect behavior.

Although maybe it would make sense to create a toast message on those that failed so there is some UI feedback here.